### ISSUE TITLE
BUG: Missing create.yml, destroy.yml, molecule_vagrant.py

### ISSUE THREAD

**Issue by westurner** [ansible-community/molecule-vagrant#8]
CC'ing from "Removed migration and vagrant driver" https://github.com/ansible-community/molecule/pull/2414#issuecomment-582297253 https://github.com/ansible-community/molecule/issues/2315 :

> When I just copy create.yml and destroy.yml from the links above into the scenario directory, 

```bash
cd ./molecule/default
wget 'https://raw.githubusercontent.com/ansible-community/molecule/0cf231a3623e10456fbc21795ae42286ed88e479/molecule/provisioner/ansible/playbooks/vagrant/destroy.yml' 
wget 'https://raw.githubusercontent.com/ansible-community/molecule/0cf231a3623e10456fbc21795ae42286ed88e479/molecule/provisioner/ansible/playbooks/vagrant/create.yml' 
```

> I'm now getting:
```
ERROR! couldn't resolve module/action 'molecule_vagrant'. This often indicates a misspelling, missing collection, or incorrect module path.
```
> presumably because `molecule_vagrant.py` is not in https://github.com/ansible-community/molecule-vagrant ?
> https://github.com/ansible-community/molecule/blob/0cf231a3623e10456fbc21795ae42286ed88e479/molecule/provisioner/ansible/plugins/modules/molecule_vagrant.py

> https://github.com/ansible-community/molecule/pull/2414/files#diff-9b0cc6dd3c61a1aca9b9482e06c74b0f

> So, to workaround this with master:
```bash
mkdir -p ~/.ansible/plugins/modules  # any dir in ANSIBLE_LIBRARY (as seen in --debug)
cd ~/.ansible/plugins/modules;
wget https://raw.githubusercontent.com/ansible-community/molecule/0cf231a3623e10456fbc21795ae42286ed88e479/molecule/provisioner/ansible/plugins/modules/molecule_vagrant.py
```

Where should these be added in molecule-vagrant?

**Comment by sio** [comment]
I confirm that the following files are missing after splitting molecule-vagrant into a plugin:

```
# File names are from molecule 2.22 source tree
molecule/provisioner/ansible/playbooks/vagrant/create.yml
molecule/provisioner/ansible/playbooks/vagrant/destroy.yml
molecule/provisioner/ansible/plugins/libraries/molecule_vagrant.py
```

I'm not sure where to add them though.

**Comment by hswong3i** [comment]
```
# File names are from molecule 2.22 source tree
molecule/provisioner/ansible/playbooks/vagrant/create.yml
molecule/provisioner/ansible/playbooks/vagrant/destroy.yml
molecule/provisioner/ansible/plugins/libraries/molecule_vagrant.py
```

These 3 files should able to place under your molecule scenario folder as a dirty workaround, e.g.
```
molecule/default/create.yml
molecule/default/destroy.yml
molecule/default/libraries/molecule_vagrant.py
```

@ssbarnea As a long run chould we create another PR for this?

**Comment by ssbarnea** [comment]
I will be back at this as soon I sort molecule v3, probably during next weekend.

**Comment by hswong3i** [comment]
@ssbarnea anyway, any hints where the `libraries/molecule_vagrant.py` should locate in this repo?

I could expect `create.yml` and `destroy.yml` copy from v2.20 into:
* https://github.com/ansible-community/molecule-vagrant/tree/master/molecule_vagrant/cookiecutter/scenario/driver/vagrant/%7B%7Bcookiecutter.molecule_directory%7D%7D/%7B%7Bcookiecutter.scenario_name%7D%7D

But I have no idea where `libraries/molecule_vagrant.py` should goes...

**Comment by hswong3i** [comment]
In case of Travis CI + Molecule + Vagrant + Libvirt + KVM + CentOS/openSUSE/Ubuntu setup, you may reference my initial changes for upgrade from 2.22.0 to 3.0.1:
  - https://github.com/alvistack/ansible-role-virtualbox/commit/4e414fc2731b23c6008de1266f6fc16c10dd7c25
  - https://travis-ci.org/alvistack/ansible-role-virtualbox/jobs/653456732

**Comment by sio** [comment]
@hswong3i, cool! I didn't know Travis supported nested virtualization

**Comment by geerlingguy** [comment]
@hswong3i you're able to run VirtualBox inside Travis CI??? I tried many times throughout the years and never could get it working.

**Comment by sio** [comment]
@geerlingguy, I believe he is running kvm (via vagrant-libvirt) to test the role that installs Virtualbox, not the Virtualbox itself.

**Comment by hswong3i** [comment]
@geerlingguy @sio 

Since @ssbarnea mention that we may try for libvirt, I give a try and so SUDDENLY found that Travis CI now coming with vmx from /prof/cpuinfo therefore both KVM or VirtualBox now works (but I  like libvirt more when compare the works for adding additional disk for Ceph test)

Moreover I found that other else github project report that since 2019-08 vmx is UNOFFICIALLY added into GCE instance, that’s also why when I tested with molecule + VirtualBox on early 2019 but still failed :-)

**Comment by jonashackt** [comment]
Wow thanks so much @hswong3i - got Vagrant to run on TravisCI (with libvirt & KVM), this is absolutely amazing - I had to create my own example project to see it in action: https://github.com/jonashackt/vagrant-travisci-libvrt What did not work until is the VirtualBox version of that setup (see https://github.com/jonashackt/vagrant-travisci) - are you really sure, VirtualBox will also work?

Either way, with that setup together with the `generic` Vagrant Boxes from Vagrant Cloud that support both `libvirt` __and__ `virtualbox` providers at the same time, a really nice possibility to work with Vagrant locally and on CloudCI jobs is available finally!

**Comment by ssbarnea** [comment]
I released v0.2 which is fixing most of the known issues, including this.

**Comment by westurner** [comment]
Thanks!
